sop-core/src/Data/SOP/NP.hs

```diff
@@ -130,12 +143,40 @@ import Data.SOP.Sing
 -- > K 0      :* K 1     :* Nil  ::  NP (K Int) '[ Char, Bool ]
 -- > Just 'x' :* Nothing :* Nil  ::  NP Maybe   '[ Char, Bool ]
 --
-data NP :: (k -> Type) -> [k] -> Type where
-  Nil  :: NP f '[]
-  (:*) :: f x -> NP f xs -> NP f (x ': xs)
+newtype NP (f :: k -> *) (xs :: [k]) = NP (V.Vector Any)
```

Comment:
I'm still unconvinced that `Any` is better than `f Any` here. Do you have any data points on this?

Reply:
I started out with just `Any` on my branch, despite having seen you use `f Any`, because I didn't see the point. Then later on I realized that actually having that `f` there is really rather helpful; often that `f` is concrete (say, an applicative functor or something) and having, say, `f (Any -> Any)` and `f Any` in hand is a lot more helpful than `Any` and `Any`.

Reply:
(Well, maybe not _concrete concrete_, but at least we know something about the structure.)